packages/apidom-ns-openapi-2/src/refractor/visitors/generics/AlternatingVisitor.ts

```diff
@@ -1,32 +1,34 @@
-import stampit from 'stampit';
 import { ifElse, always } from 'ramda';
 import { dispatch, stubUndefined } from 'ramda-adjunct';
 import { Element, BREAK } from '@swagger-api/apidom-core';
 
-import SpecificationVisitor from '../SpecificationVisitor';
-
-const AlternatingVisitor = stampit(SpecificationVisitor, {
-  props: {
-    alternator: [],
-  },
-  methods: {
-    enter(element: Element) {
-      const functions = this.alternator.map(
-        ({
-          predicate,
-          specPath,
-        }: {
-          predicate: (element: unknown) => boolean;
-          specPath: string[];
-        }) => ifElse(predicate, always(specPath), stubUndefined),
-      );
-      const specPath = dispatch(functions)(element);
-
-      this.element = this.toRefractedElement(specPath, element);
-
-      return BREAK;
-    },
-  },
-});
+import SpecificationVisitor, { SpecificationVisitorOptions } from '../SpecificationVisitor';
+
+export type Alternator = { predicate: (element: unknown) => boolean; specPath: string[] };
+
+export interface AlternatingVisitorOptions extends SpecificationVisitorOptions {
+  readonly alternator: Alternator[];
```

Comment:
If it's not required because of `this.alternator = alternator || [];`, let's use optional marker. 

```suggestion
  readonly alternator?: Alternator[];
```